app/models/spree/order.rb

```diff
@@ -756,11 +756,19 @@ def associate_customer
     def ensure_customer
       return if associate_customer
 
-      customer_name = bill_address.andand.full_name
-      self.customer = Customer.create(enterprise: distributor, email: email_for_customer,
-                                      user: user, name: customer_name,
-                                      bill_address: bill_address.andand.clone,
-                                      ship_address: ship_address.andand.clone)
+      self.customer = Customer.new(
+        enterprise: distributor,
+        email: email_for_customer,
+        user: user,
+        name: bill_address.andand.full_name,
+        bill_address: bill_address.andand.clone,
+        ship_address: ship_address.andand.clone
+      )
+      customer.save
+
+      if customer.invalid?
```

Comment:
Why not check `#persisted?`. I think it avoids a query (and it's more explicitly what we're looking to check, righ?)...

```ruby
irb(main):006:0> customer.persisted?
=> false
irb(main):007:0> customer.valid?
  Spree::Customer Exists (1.6ms)  SELECT  1 AS one FROM "spree_customers" WHERE "spree_customers"."email" = $1 LIMIT $2  [["email", "tim@test.com"], ["LIMIT", 1]]
=> false
irb(main):008:0> customer.invalid?
  Spree::Customer Exists (1.3ms)  SELECT  1 AS one FROM "spree_customers" WHERE "spree_customers"."email" = $1 LIMIT $2  [["email", "tim@test.com"], ["LIMIT", 1]]
=> true
```

Reply:
but `invalid?` triggers the model validations (AFAIK persisted? does not) and it's more intention revealing. Besides, we aiming to get this code out to assess the situation. It's but no means a solution.

Reply:
I find more intention revealing to check for valid?/validation?. What we care for is whether or not the customer passed the validations and if not, we'll list which ones failed.

Besides, this is by no means a solution. We're just trying to get some more insights but I'll see saves unnecessary queries anway.